linera-views/src/scylla_db.rs

```diff
@@ -317,18 +371,126 @@ impl ScyllaDbClient {
     }
 }
 
+/// A shared DB client for ScyllaDb implementing LruCaching
+#[derive(Clone)]
+pub struct ScyllaDbClient {
+    client: LruCachingKeyValueClient<ScyllaDbClientInternal>,
+}
+
+#[async_trait]
+impl KeyValueStoreClient for ScyllaDbClient {
+    const MAX_VALUE_SIZE: usize = ScyllaDbClientInternal::MAX_VALUE_SIZE;
+    type Error = <ScyllaDbClientInternal as KeyValueStoreClient>::Error;
+    type Keys = <ScyllaDbClientInternal as KeyValueStoreClient>::Keys;
+    type KeyValues = <ScyllaDbClientInternal as KeyValueStoreClient>::KeyValues;
+
+    fn max_stream_queries(&self) -> usize {
+        self.client.max_stream_queries()
+    }
+
+    async fn read_key_bytes(&self, key: &[u8]) -> Result<Option<Vec<u8>>, Self::Error> {
+        self.client.read_key_bytes(key).await
+    }
+
+    async fn read_multi_key_bytes(
+        &self,
+        key: Vec<Vec<u8>>,
+    ) -> Result<Vec<Option<Vec<u8>>>, Self::Error> {
+        self.client.read_multi_key_bytes(key).await
+    }
+
+    async fn find_keys_by_prefix(&self, key_prefix: &[u8]) -> Result<Self::Keys, Self::Error> {
+        self.client.find_keys_by_prefix(key_prefix).await
+    }
+
+    async fn find_key_values_by_prefix(
+        &self,
+        key_prefix: &[u8],
+    ) -> Result<Self::KeyValues, Self::Error> {
+        self.client.find_key_values_by_prefix(key_prefix).await
+    }
+
+    async fn write_batch(&self, batch: Batch, base_key: &[u8]) -> Result<(), Self::Error> {
+        self.client.write_batch(batch, base_key).await
+    }
+
+    async fn clear_journal(&self, base_key: &[u8]) -> Result<(), Self::Error> {
+        self.client.clear_journal(base_key).await
+    }
+}
+
+impl ScyllaDbClient {
+    /// Get the table_name of a client
+    pub async fn get_table_name(&self) -> String {
+        self.client.client.get_table_name().await
+    }
+
+    /// Creates a [`ScyllaDbClient`] from the input parameters.
+    pub async fn new(
+        restart_database: bool,
+        uri: &str,
+        namespace: String,
+        max_concurrent_queries: Option<usize>,
+        max_stream_queries: usize,
+        cache_size: usize,
+    ) -> Result<Self, ScyllaDbContextError> {
+        let client = ScyllaDbClientInternal::new(
+            restart_database,
+            uri,
+            namespace,
+            max_concurrent_queries,
+            max_stream_queries,
+        )
+        .await?;
+        let client = LruCachingKeyValueClient::new(client, cache_size);
+        Ok(ScyllaDbClient { client })
+    }
+}
+
+/// Get a test table_name
```

Comment:
```suggestion
/// Gets a test table name
```

Or maybe: "Returns a unique table name for testing."

Reply:
Yes, your rewrite is much better.